packages/pds/tests/app-passwords.test.ts

```diff
@@ -0,0 +1,103 @@
+import AtpAgent from '@atproto/api'
+import * as jwt from 'jsonwebtoken'
+import { CloseFn, runTestServer, TestServerInfo } from './_util'
+
+describe('app_passwords', () => {
+  let server: TestServerInfo
+  let accntAgent: AtpAgent
+  let appAgent: AtpAgent
+  let close: CloseFn
+
+  beforeAll(async () => {
+    server = await runTestServer({
+      dbPostgresSchema: 'app_passwords',
+    })
+    accntAgent = new AtpAgent({ service: server.url })
+    appAgent = new AtpAgent({ service: server.url })
+    close = server.close
+
+    await accntAgent.createAccount({
+      handle: 'alice.test',
+      email: 'alice@test.com',
+      password: 'alice-pass',
+    })
+  })
+
+  afterAll(async () => {
+    await close()
+  })
+
+  let appPass: string
+
+  it('creates an app-specific password', async () => {
+    const res = await accntAgent.api.com.atproto.server.createAppPassword({
+      name: 'test-pass',
+    })
+    expect(res.data.name).toBe('test-pass')
+    appPass = res.data.password
+  })
+
+  it('creates a session with an app-specific password', async () => {
+    const res = await appAgent.login({
+      identifier: 'alice.test',
+      password: appPass,
+    })
+    expect(res.data.did).toEqual(accntAgent.session?.did)
+  })
+
+  it('creates an access token for an app with a restricted scope', () => {
+    const decoded = jwt.decode(appAgent.session?.accessJwt ?? '', {
+      json: true,
+    })
+    expect(decoded?.scope).toEqual('com.atproto.appPass')
+  })
+
+  it('allows actions to be performed from app', async () => {
+    await appAgent.api.app.bsky.feed.post.create(
+      {
+        repo: appAgent.session?.did,
+      },
+      {
+        text: 'Testing testing',
+        createdAt: new Date().toISOString(),
+      },
+    )
+  })
+
+  it('restricts certain actions', async () => {
```

Comment:
Just double-checking that `createAppPassword` is the only method we want to restrict in this PR.

Reply:
I think so? I can't think of any others besides maybe `revokeAppPassword`

Honestly I'm not even sure we need to restrict `createAppPassword` 🤔 

`deleteAccount` requires you to add in your account password anyways

Reply:
I think it's a good call to restrict `createAppPassword` 👍